core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -292,8 +288,12 @@ private[spark] class TaskSetManager(
   {
     speculatableTasks.retain(index => !successful(index)) // Remove finished tasks from set
 
-    def canRunOnHost(index: Int): Boolean =
-      !hasAttemptOnHost(index, host) && !executorIsBlacklisted(execId, index)
+    def canRunOnHost(index: Int): Boolean = {
+      !hasAttemptOnHost(index, host) && blacklistTracker.map { bl =>
```

Comment:
I don't know, the indentation here bothers me a bit, makes the statement confusing to read. Maybe moving `blacklistTracker.map { bl =>` to the next line and indenting it would help.